src/app/modules/examples/examples.utils.ts

```diff
@@ -0,0 +1,224 @@
+ * @param groupByMiddleware The inisital The id of the agency to filter forms listed by. If no id is given, all matched forms will be returned
+ * @param agencyId Optional. The id of the agency to filter forms listed by. If no id is given, all matched forms will be returned
+ * @param minSubmissionCount The minimum submission count returned forms must have to be returned by the aggregation pipeline
+ */
+export const createGeneralQueryPipeline = ({
+  groupByMiddleware,
+  agencyId,
+  minSubmissionCount,
+}: {
+  groupByMiddleware: Record<string, unknown>[]
+  agencyId?: string
+  minSubmissionCount: number
+}): Record<string, unknown>[] => {
+  return groupByMiddleware.concat(
+    // Only display forms with more than minSubmissionCount submissions.
+    filterBySubmissionCount(minSubmissionCount),
+    // More recently submitted forms appear higher on the examples page.
+    sortByLastSubmitted,
+    // Retrieve form info from the forms that reach this step.
+    lookupFormInfo,
+    // Filter out all inactive/unlisted forms.
+    filterInactiveAndUnlistedForms,
+    // Retrieve agency infos of forms in this stage.
+    lookupAgencyInfo,
+    // Filter by agency id if parameter given.
+    agencyId ? filterByAgencyId(agencyId) : [],
+    // Retrieve form feedback from the forms that reach this step.
+    lookupFormFeedback,
+  )
+}
+
+/**
+ * Creates a query pipeline that can be used to retrieve the example card data
+ * for the /examples page with the specified form id.
+ * This pipeline only provides formId, form title, agency logo and color theme.
+ * @param formId The specific form to query.
+ */
+export const createFormIdInfoPipeline = (
+  formId: string,
+): Record<string, unknown>[] => {
+  return searchFormsById(formId).concat(
+    filterInactiveAndUnlistedForms,
+    lookupAgencyInfo,
+    projectFormDetails,
+  )
+}
+
+/**
+ * Creates a query pipeline that can be used to retrieve a single example form
+ * for the /examples page using the submission collection.
+ *
+ * This pipeline will return the average feedback for the form id referenced to
+ * be shown in the example form.
+ * @param formId. The id of the form to retrieve data for
+ */
+export const createSingleSearchSubmissionPipeline = (
+  formId: string,
+): Record<string, unknown>[] => {
+  return searchSubmissionsForForm('form', formId).concat(
+    sortByCreated,
+    groupSubmissionsByFormId,
+    sortByLastSubmitted,
+    lookupFormFeedback,
+    addAvgFeedback,
+  )
+}
+
+/**
+ * Creates a query pipeline that can be used to retrieve a single example form
+ * for the /examples page using the formStatisticsTotal collection.
+ *
+ * This pipeline will return the average feedback for the form id referenced to
+ * be shown in the example form.
+ * @param formId. The id of the form to retrieve data for
+ */
+export const createSingleSearchStatsPipeline = (
+  formId: string,
+): Record<string, unknown>[] => {
+  return searchSubmissionsForForm('formId', formId).concat(
```

Comment:
add a comment why the key is `formId`?

Reply:
I'm actually not sure. I've just refactored directly from the old code. Care to enlighten?

Reply:
https://github.com/datagovsg/formsg-batch-jobs/blob/dc0b9b4fbad7a17287f445b84cf63cbc72505993/formSubmissionStatisticsJob/src/batchJobs/form-submission-statistics-job.js#L205-L224

Reply:
haha we had a discussion about this remember? How form stats total schema is not declared anywhere

Reply:
I guess add a comment and link to the batch jobs code above so the next developer doesn't need to figure it out?

Reply:
As in, this function is searching submissions collection and not the form stats collection, is the query wrong?

Reply:
aaahh, the comment says _formStatisticsTotal_?

Reply:
Oh my goodness, I self confused myself. Yes, `formId` is correct since the aggregation pipeline this will run on will be on the FormStatisticsTotal collection.... This shows how confusing these example queries are lol